src/appsettings/settings.py

```diff
@@ -1316,3 +1429,21 @@ def check(self):
     def transform(self, value):
         """Transform each item in the list."""
         return tuple(self.inner_setting.transform(item) for item in value)
+
+    def decode_environ(self, value):
+        """
+        Impossible to decode more complex types from environment variables, raise an error.
+
+        Args:
+            value (str):
+
+        Raises:
+            ValueError: Impossible to decode a string variable to a complex data type by default.
+
+        Returns:
+            None:
+        """
+        raise ValueError(
+            "Environment value for {} has been set, but no decode method implemented for this complex "
+            "data type".format(self.full_name)
+        )
```

Comment:
This shouldn't be much harder to decode than `ListSetting`. Don't you want to try and create a decoder?

Reply:
I tried to do it, split by `:` and then `=`. The only problem is, that it is not possible to create a list inside of a dict this way...

Reply:
I'd support simple space separated `key=value` pairs for simple cases and JSON for more complex structures.

Reply:
Done.